src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -239,165 +287,230 @@ export default class ExtensionRequest extends Vue {
     }
   }
 
-  /** Whether to disable the editing of AGM Year field. */
-  get isFirstAgm (): boolean {
-    // Field disabled on first load when value is null
-    if (this.data.isFirstAgm == null) {
-      return true
-    }
-
-    // Field disabled when it is first agm
-    return this.data.isFirstAgm
+  /** Incorporation date as text. */
+  get incorporationDateText (): string {
+    return DateUtilities.dateToYyyyMmDd(
+      this.data?.incorporationDate
+    )
   }
 
   /** Called when isFirstAgm radio group changes. */
   @Watch('data.isFirstAgm')
   onIsFirstAgmChanged (val: boolean): void {
     if (val) {
       this.data.agmYear = this.data.incorporationDate.getFullYear().toString()
+      this.data.prevAgmDate = null
     } else {
       this.data.agmYear = ''
     }
   }
 
-  /** Called when extension date picker changes. */
-  @Watch('extensionExpiryDateText')
-  onExtensionDatePickerChanged (val: string): void {
-    this.data.prevExpiryDate = val
+  /** Called when isPrevExtension radio group changes. */
+  @Watch('data.isPrevExtension')
+  onIsPrevExtensionChanged (val: boolean): void {
+    if (!val) {
+      this.data.prevExpiryDate = null
+    }
   }
 
-  /** Called when intended AGM date picker changes. */
-  @Watch('intendedAgmDateText')
-  onAgmIntendedDatePickerChanged (val: string): void {
-    this.data.intendedAgmDate = val
+  /** Watches all data properties to keep track of changes. */
+  @Watch('data.isFirstAgm')
+  @Watch('data.prevAgmDate')
+  @Watch('data.isPrevExtension')
+  @Watch('data.prevExpiryDate')
+  @Watch('data.intendedAgmDate')
+  @Watch('data.currentDate')
```

Comment:
Would it make sense to use `@Watch('data', { deep: true })` here, or are there data properties you don't want to watch?

Reply:
I don't think so, since when this watcher triggers the `checkEligibility()` function, we are setting a bunch of data properties like `isElgibile`, `extensionDuration`, `requestExpired`, etc. that we don't necessarily want to watch for changes to.